src/main/kotlin/pt/up/fe/ni/website/backend/controller/PostController.kt

```diff
@@ -18,18 +18,21 @@ class PostController(private val service: PostService) {
     @GetMapping
     fun getAllPosts() = service.getAllPosts()
 
-    @GetMapping("/{postId}")
+    @GetMapping("/{postId:[0-9]+}")
     fun getPost(@PathVariable postId: Long) = service.getPostById(postId)
 
     @PostMapping("/new")
     fun createPost(@RequestBody dto: PostDto) = service.createPost(dto)
 
-    @PutMapping("/{postId}")
+    @PutMapping("/{postId:[0-9]+}")
     fun updatePost(
         @PathVariable postId: Long,
         @RequestBody dto: PostDto
     ) = service.updatePostById(postId, dto)
 
-    @DeleteMapping("/{postId}")
+    @DeleteMapping("/{postId:[0-9]+}")
     fun deletePost(@PathVariable postId: Long) = service.deletePostById(postId)
+
+    @GetMapping("/{postSlang:[a-z-]+}")
```

Comment:
With this regex, we cannot create slugs containing numbers (or any characters except lowercase letters). I think we should this endpoint to handle all requests not being handled by `getById`. That way, we always get a semantic error like "no slang found" instead of "invalid endpoint". I can think of two ways of doing this:
- Use a wildcard, `@GetMapping("/{postSlang}**")`, to lower specificity (see [docs](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-ann-requestmapping-pattern-comparison))
- Negate the regex of the id